src/components/TreeGrid/TreeGrid.Props.ts

```diff
@@ -0,0 +1,28 @@
+import { GridColumn, SortDirection } from '../QuickGrid/QuickGrid.Props';
+
+export interface ITreeGridProps {
+    tree: Array<TreeNode>;
+    columns: Array<GridColumn>;
+    className?: string;
+    onRowDoubleClicked?: (row: any) => void;
+    onSelectedRowChanged?: (selectedRowIndex: number) => void;
+    sortColumn?: string;
+    sortDirection?: SortDirection;
+    columnSummaries?: any;
+}
+
+export interface ITreeGridState {
+    collapsedTreeNodes: Array<TreeNode>;
+    selectedRowIndex?: number;
+    columnsToDisplay: Array<GridColumn>;
+    sortColumn?: string;
+    sortDirection?: SortDirection;
+}
+
+
+export interface TreeNode { // extend this interface on a data structure to be used for row data
+    TreeId: string;
```

Comment:
Is this the Id of the node or the id of the whole tree? I presume just the node. In that case it would be better to have id as the name.

Reply:
Also we typically use camelCase convention for property naming. Can you please change all the properties to camelCase?

Reply:
Changed